packages/binding-coap/src/coap-client.ts

```diff
@@ -231,11 +238,47 @@ export default class CoapClient implements ProtocolClient {
         return defaultMethod;
     }
 
+    private setBlockOption(req: OutgoingMessage, optionName: "Block1" | "Block2", blockSZX?: BlockSize): void {
+        if (blockSZX == null) {
+            return;
+        }
+
+        try {
+            const block2OptionValue = blockSizeToOptionValue(blockSZX);
+            req.setOption(optionName, block2OptionValue);
+        } catch (e) {
+            warn(e.toString());
+        }
+    }
+
+    private applyFormData(form: CoapForm, req: OutgoingMessage) {
+        // TODO: Add qblock parameters and hop limit, once implemented in node-coap
+        // TODO: Better differentiate between content type and accept
+
+        const contentFormat = form.contentType;
+        if (contentFormat != null) {
+            debug(`"CoapClient got Form 'contentType' ${contentFormat} `);
+            req.setOption("Content-Format", contentFormat);
```

Comment:
Let's say there is a property that looks like this:

```json
{
    "op": "readproperty",
    "cov:method": "GET",
    "contentType": "application/json"
}
```

Wouldn't this code set the `Content-Format` option in the GET request to `50`, implying erroneously that the GET request has a payload?

Reply:
Also, since the `contentType` member of a form is mandatory with default value "application/json", I guess the code should look like this:

```diff
-const contentFormat = form.contentType;
-if (contentFormat != null) {
+const contentFormat = form.contentType || "application/json";
 req.setOption("Content-Format", contentFormat);
```

This would mean that *every* request (regardless of method) would have to have a payload, no?

Reply:
Thank you for the feedback :) I added a couple of updates to the PR. Currently, however, there are a few limitations in node-coap which prevent me from using the numeric values from `cov:contentFormat` and `cov:accept`. I'll try to add these features to node-coap in the upcoming days, then hopefully this PR will be able to be finished :)

Reply:
I updated node-coap in the meantime, so this thread can be considered resolved :)